README.md

```diff
@@ -0,0 +1,60 @@
+# Introduction
+
+The **hal_gigadevice** is a set of standard firmware library from GigaDevice.
+
+# How to submit code
+
+ - When submiting an updated firmware library version, it is important make
+ sure that the last firmware version will be merged. The library version can
+ be checked at [GigaDevice MCU Site](http://www.gd32mcu.com/en/download/).
+ - This repository is managed mainly on Linux. This means that changes should
+ be submited using Linux LF format. Usually running a similar command to
+ `find * type f -print0 | xargs -0 dos2unix -k` should be enough.
+ - Make sure directory structure is respected.
+ - Directory names should be converted to lowercase.
+ - All public API funtions must be prefixed with `gd32_`.
+ - All public GigaDevices related defines and enum values must be prefixed
+   with `GD32_`.
```

Comment:
Can we reserve the prefix gd32_ to driver or kernel developer? They can use these prefix freely with no name conflict risk. That's why I agree with the prefix "__". 
May be gd32_lib_ or gd32_hal_?

Reply:
Hi @cameled ,

There is no need to reserve prefix for Zephyr core/drivers. The usual naming convention is `<driver>_<manufacturer>_funcX`. For instance, at USART driver we have `usart_gd32_funcX`. So, it is safe only use `gd32_ ` wide spread at hal_gigadevice.

Reply:
In some cases `GD32_` is used, see https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/usart_gd32.c#L102

Reply:
> In some cases GD32_ is used, see https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/usart_gd32.c#L102

Yes, correctly. However, if we agreed that prefixes are the solution, function names are automatically excluded from conflict list due to `<driver>_<manufacturer>_funcX` standard syntax in drivers and vendor prefixes are not used and should not be used at Zephyr core and libraries. In the case of enum/macros, only newer conflicts can happen at development phase and developer will get a error. In this case, developer should select another name.